src/service-providers/decision-trees/decision-trees.adapters.ts

```diff
@@ -11,3 +18,67 @@ export const adaptDtreeStatResponse = (
     totalCounts: adaptFilteringStatsCounts(response['total-counts']),
   }
 }
+
+export const adaptDtreeSetResponse = (
+  response: IDtreeSetResponse,
+): IDtreeSet => {
+  const initialStep: IStepData = {
+    step: 1,
+    groups: [],
+    excluded: true,
+    isActive: false,
+    isReturnedVariantsActive: false,
+    conditionPointIndex: 0,
+    returnPointIndex: null,
+  }
+
+  const stepCodes = getDataFromCode(response.code)
+
+  const localSteps: IStepData[] = []
+  const atomsEntries = Object.entries(response['cond-atoms'] ?? {})
+
+  atomsEntries.forEach(([key, atom], index) => {
+    const conditionPointIndex = parseInt(key, 10)
+
+    localSteps.push({
+      step: index + 1,
+      groups: atom.filter((elem: any[]) => elem.length > 0),
+      excluded: !stepCodes[index].result,
+      isActive: false,
+      isReturnedVariantsActive: false,
+      conditionPointIndex,
+      returnPointIndex: conditionPointIndex + 1,
+      comment: stepCodes[index].comment,
+      negate: stepCodes[index].isNegate,
+      all: stepCodes[index].isAll,
+      condition: stepCodes[index].condition,
+    })
+  })
+
+  localSteps.forEach((step: IStepData, index: number) => {
```

Comment:
@Lapk1n removed this in his PR.
And also, he refactored some fields in `localSteps`